### PR TITLE
hotfix -  script migrating old passages to store in sub directory

### PR DESCRIPTION
**Pull request by pnal**
In order to fix [rich passage authoring bug](https://oat-sa.atlassian.net/browse/AUT-1620) we need to run this script on the environment.
As discussed we going to test it on some env and then include it as a migration.
Short how-to use document: https://oat-sa.atlassian.net/wiki/spaces/~437715569/pages/1709211880/AUT-1620+Fix+passages+created+before+rich+passage+authoring+was+implemented

### CODE REVIEW

File: scripts/ReplaceSharedStimulusMedia.php [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+    public function __invoke($params)
+    {
+        $mediaInstances = $this->getMediaService()->getRootClass()->getInstances(true);
+
+        $report = Report::createInfo(__('%s media on this environment', count($mediaInstances)));
+        $sharedStimulusCount = 0;
+        $toBeChangedCount = 0;
+        $successCount = 0;
+
+        $dryRun = isset($params[0]) && strpos($params[0], 'dryRun') !== false;
```
Comment (andreluizmachado) [inline scripts/ReplaceSharedStimulusMedia.php:61]: Why are you using strpos? why do not we compare the entire string?

File: scripts/ReplaceSharedStimulusMedia.php [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+use oat\taoMediaManager\model\sharedStimulus\specification\SharedStimulusResourceSpecification;
+
+/**
+ * Used to update old media from the command line
+ *
+ * ```
+ * sudo -u www-data php index.php 'oat\taoMediaManager\scripts\ReplaceSharedStimulusMedia' [dryRun]
+ * ```
+ */
+class ReplaceSharedStimulusMedia extends ConfigurableService implements Action
```
Comment (andreluizmachado) [inline scripts/ReplaceSharedStimulusMedia.php:44]: can we add automated tests for this class?

File: scripts/ReplaceSharedStimulusMedia.php [hunk 3]
```diff
@@ -0,0 +1,158 @@
+<?php
```
Comment (andreluizmachado) [inline scripts/ReplaceSharedStimulusMedia.php:1]: shouldn't we create a migration for this script?

**Review by andreluizmachado — CHANGES_REQUESTED** [review]

File: scripts/ReplaceSharedStimulusMedia.php (hunk 3 above)
Comment (andreluizmachado) [reply scripts/ReplaceSharedStimulusMedia.php:1]: please check: https://oat-sa.atlassian.net/browse/AUT-1620?focusedCommentId=165847

[the remark this answers is not in the document]
Comment (andreluizmachado) [reply scripts/ReplaceSharedStimulusMedia.php:1]: please check: https://oat-sa.atlassian.net/browse/AUT-1620?focusedCommentId=165847

File: scripts/ReplaceSharedStimulusMedia.php (hunk 2 above)
Comment (andreluizmachado) [reply scripts/ReplaceSharedStimulusMedia.php:44]: Please check: https://oat-sa.atlassian.net/browse/AUT-1620?focusedCommentId=165847

File: scripts/ReplaceSharedStimulusMedia.php (hunk 1 above)
Comment (andreluizmachado) [reply scripts/ReplaceSharedStimulusMedia.php:61]: Please check: https://oat-sa.atlassian.net/browse/AUT-1620?focusedCommentId=165847

**Review by andreluizmachado — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [ ] New code is not subject to concurrency issues (if applicable)
- [ ] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [ ] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

**Review by bziondik — APPROVED** [review]
LGTM

**Pull request merged**